server/rpc.go

```diff
@@ -52,6 +52,45 @@ type Broadcaster interface {
 	Sign([]byte) ([]byte, error)
 }
 
+// Balance describes methods for a session's balance maintenance
+type Balance interface {
```

Comment:
I prefer to use an interface here because it makes testing here easier by abstracting the implementation details for balance maintenance away from `BroadcastSession`. If we use a concrete type here, then `BroadcastSession` is made aware of how balances are actually maintained under the hood and these details seep into how we write our tests for `SubmitSegment()` as well.

Reply:
I also considered defining `BalanceUpdate` in `core/accounting.go`, but the type didn't seem to have much relevance to the implementation as it would just be wrapping values returned by `StageUpdate()`. The real utility of the type after being created by `StageUpdate()` is to keep track of the status of an update which changes during the execution of `SubmitSegment()`. So, given that the status of a balance update is defined within `SubmitSegment()`, I felt that it would be ok defining `BalanceUpdate` in `server/rpc.go`